mvc/models/model-binding.rst

```diff
@@ -1,8 +1,105 @@
+
+`http://contoso.com/movies/edit/2`
+
+The default model binder compares the sections of the URL with the pattern defined in the ``RouteValues`` collection. Put a breakpoint in any action method to see the contents of the ``RouteData`` object at run time in a debug window, as illustrated here: 
+
+.. image:: model-binding/_static/routevalues-in-watch-window.png
+
+Since the  routing pattern looks like this, ``{controller=Home}/{action=Index}/{id?}`` (it's the default route), the route parts ``movies/edit/2`` map to the ``Movies`` controller, its ``Edit`` action method, which accepts an argument called ``id``. So the code for the action method should look something like this: 
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/Controllers/MoviesController.cs
+   :language: c#
+   :lines: 63-76  
+   
+.. note:: The strings in the URL are not case sensitive.
+
+Now that routing knows which action method to use, it quickly checks for model binding attributes applied to the action method's arguments. If it finds a model binding attribute it will use the data source specified by the attribute. Otherwise, for each argument (a primitive type for now, more on complex types later), it looks in its data sources for something with a matching name. In the above example the only argument is an integer named `id`, so that is what the binder will look for. Below is a list of the data sources in the order that the model binder looks through them:
+ 
+#. ``Request.Form``: These are form values that go in the HTTP request using the POST method.
+#. ``RouteData.Values``: Contains information about each segment of the route. Route segments are separated by a forward slash ``/``.
+#. ``Request.QueryStrings``: Data to the right of a question mark (?) in a URL, separated by ampersands (&). For example, genre and subgenre are query strings: http://contoso.com/movies?genre=Drama&subgenre=Crime
+
+.. note:: Form, RouteData, and QueryStrings are all stored as name-value pairs.
+
+Web Developers work with form and querystring data regardless of the Web server or development environment and languages, so these should be familiar. In the above example, the model binder is will start looking through the Request.Form first for a key named `id`. In this case it didn't find anything, so it moved on to RouteData.Values looking for a key with a name of `id`. In our example, it's a match, since the last portion of the route has a key value of `id`, and it is an integer. Binding happens, and the value of the id parameter becomes 2. If the action method's parameter were a complex type such as the ``Movie`` type, which likely contains both primitive and complex types, MVC's model binder will still handle it nicely. It uses reflection and recursion to traverse the properties of complex types through its tree, and comparing any primitive members' names against the names of keys in data sources. The model binder can bind objects that contain nested objects for multiple levels. 
```

Comment:
can we delete "Web Developers work with form and querystring data regardless of the Web server or development environment and languages, so these should be familiar. "

binder is will <-- remove is
I prefer example above but that's my style
I think you could remove above
In the example, the binder looks (remove start)